data/console/test_java.sh

```diff
@@ -197,81 +197,90 @@ else
+        # Test if there's an alternativ for java, and if yes, set it as the current used one
+        if grep -q $dot_version $LIST_ALL_JAVA_ALTERNATIVES; then
+	    java=$(grep $dot_version $LIST_ALL_JAVA_ALTERNATIVES)
+	    update-alternatives --set java $java
+        else
+	    echo "Error: java alternative not found for $java_version"
+	    exit 1
+        fi
+        # Test if there's an alternative for javac, and if yes, set it as the current used one
+        if grep -q $dot_version $LIST_ALL_JAVAC_ALTERNATIVES; then
+	    javac=$(grep $dot_version $LIST_ALL_JAVAC_ALTERNATIVES)
+	    update-alternatives --set javac $javac
+        else
+	    if echo $java_version | grep -q ibm; then
+	        if ! rpm -qa | grep 'java.*devel' | grep -q $dot_version; then
+		    echo "Warning: java compiler alternative not found for $java_version"
+		    echo "Reason : devel pkg is not installed, thus it is normal there is not javac"
+		    if zypper products -i | grep -q sle-sdk; then
+		        echo "Status : Error! The devel pkg should be installed. Check your repos!"
+		        exit 1
+		    else
+		        echo "Status : Accepted because there is no SDK installed in the SUT"
+		    fi
+		    echo "SKIP Testing $java_version"
+		    echo
+		    continue
+	        fi
+	    fi
+	    echo "Error: java compiler alternative not found for $java_version"
+	    exit 1
+        fi
+        # Test if there's an alternativ for javaplugin, and if yes, set it as the current used one
+        # So far, only java-ibm offers this
+        if [[ $(uname -m) -ne "s390x" ]]; then
+	    if echo $java_version | grep -q ibm; then
+	        if grep $dot_version $LIST_ALL_JAVAPLUGIN_ALTERNATIVES > /dev/null; then
+		    javaplugin=$(grep $dot_version $LIST_ALL_JAVAPLUGIN_ALTERNATIVES)
+		    update-alternatives --set javaplugin $javaplugin
+	        else
+		    echo "Error: java plugin alternative not found for $java_version"
+		    exit 1
+	        fi
+	    fi
         fi
-    fi
 
-    # Test version active (linked)
-    test_java_version_active
-    test_javac_version_active
-    check_version_active_vs_dot "$java_version_active" "java"
-    check_version_active_vs_dot "$javac_version_active" "javac"
+        # Test version active (linked)
+        test_java_version_active
+        test_javac_version_active
+        check_version_active_vs_dot "$java_version_active" "java"
+        check_version_active_vs_dot "$javac_version_active" "javac"
+    done
+fi
 
-    # Compile Hello World
+if [[ "$1" == "--transactional-server" ]]; then
+    echo -e "\n-----------------------------------------------------------------------------------------------------------------"
+    echo "Skip java/javac/plugin update alternative and some tests because of read-only file system on transactional server"
+    echo "-----------------------------------------------------------------------------------------------------------------"
+fi
+
+# Compile Hello World
     javac $HELLO_WORLD
```

Comment:
This has to be part of the for loop still. I recommend moving it into a function.

Reply:
yes, thanks.